multy_test/test_properties.cpp

```diff
@@ -0,0 +1,585 @@
+    EXPECT_NE(nullptr, error);
+    EXPECT_EQ(0, int_property);
+    EXPECT_NE(nullptr, &properties);
+
+    error.reset(properties_set_int32_value(&properties, nullptr, 1));
+    EXPECT_NE(nullptr, error);
+    EXPECT_EQ(0, int_property);
+    EXPECT_NE(nullptr, &properties);
+
+    error.reset(properties_set_int32_value(nullptr, "v", 1));
+    EXPECT_NE(nullptr, error);
+    EXPECT_EQ(0, int_property);
+}
+
+GTEST_TEST(PropertiesTestInvalidArgs, properties_set_string_value)
+{
+    ErrorPtr error;
+    Properties properties("");
+    std::string value_property = "2";
+
+    error.reset(properties_set_string_value(&properties, "v", "1"));
+    EXPECT_NE(nullptr, error);
+    EXPECT_NE(nullptr, &properties);
+
+    properties.bind_property("v", &value_property);
+
+    error.reset(properties_set_string_value(&properties, "v", nullptr));
+    EXPECT_NE(nullptr, error);
+    EXPECT_STREQ("2", value_property.c_str());
+    EXPECT_NE(nullptr, &properties);
+
+    error.reset(properties_set_string_value(&properties, "", "1"));
+    EXPECT_NE(nullptr, error);
+    EXPECT_STREQ("2", value_property.c_str());
+    EXPECT_NE(nullptr, &properties);
+
+    error.reset(properties_set_string_value(&properties, nullptr, "1"));
+    EXPECT_NE(nullptr, error);
+    EXPECT_STREQ("2", value_property.c_str());
+    EXPECT_NE(nullptr, &properties);
+
+    error.reset(properties_set_string_value(nullptr, "v", "1"));
+    EXPECT_NE(nullptr, error);
+    EXPECT_STREQ("2", value_property.c_str());
+}
+
+GTEST_TEST(PropertiesTestInvalidArgs, properties_set_amount_value)
+{
+    ErrorPtr error;
+    Properties properties("");
+    Amount amount_property(0);
+    Amount invalid_amount(1);
+
+    error.reset(properties_set_amount_value(&properties, "v", &invalid_amount));
+    EXPECT_NE(nullptr, error);
+    EXPECT_NE(nullptr, &properties);
+
+    properties.bind_property("v", &amount_property);
+
+    error.reset(properties_set_amount_value(&properties, "", &invalid_amount));
+    EXPECT_NE(nullptr, error);
+    EXPECT_STREQ("0", amount_property.get_value().c_str());
+    EXPECT_NE(nullptr, &properties);
+
+    error.reset(properties_set_amount_value(&properties, nullptr, &invalid_amount));
+    EXPECT_NE(nullptr, error);
+    EXPECT_STREQ("0", amount_property.get_value().c_str());
+    EXPECT_NE(nullptr, &properties);
+
+    error.reset(properties_set_amount_value(nullptr, "v", &invalid_amount));
+    EXPECT_NE(nullptr, error);
+    EXPECT_STREQ("0", amount_property.get_value().c_str());
+}
+
+GTEST_TEST(PropertiesTestInvalidArgs, properties_set_binary_data_value)
+{
+    ErrorPtr error;
+    Properties properties("");
+    const unsigned char data_4_vals[] = {1U, 2U, 3U, 4U};
+    const unsigned char data_2_vals[] = {4U,2U};
+    const BinaryData reference_value{data_4_vals, 4};
+    BinaryDataPtr binaty_data_4_property1(make_clone(BinaryData{data_4_vals, 4}));
+    BinaryDataPtr binaty_data_2_property2(make_clone(BinaryData{data_2_vals, 2}));
```

Comment:
IMO it would be better to have an empty BinaryDataPtr bound as property, and verify that no value was set.